dotnet-sonarscanner fails with duplicate ProjectGuid message if 'dotnet test' is run without the --no-build option when run in an Azure Pipeline

### Description

sonarscanner works fine with my azure pipelines until I ~~tried to turned on Code coverage... ...and~~ moved the `dotnet test` step from after the scanner `end` step to before the scanner `end` step.

My project: https://github.com/iron9light/HOCON.Json
The master works fine.
But this PR https://github.com/iron9light/HOCON.Json/pull/2 will failed.

Logs:
```
##[section]Starting: SonarScanner end
==============================================================================
Task         : Command Line
Description  : Run a command line script using cmd.exe on Windows and bash on macOS and Linux.
Version      : 2.148.0
Author       : Microsoft Corporation
Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613735)
==============================================================================
Generating script.
Script contents:
dotnet-sonarscanner end /d:sonar.login="xxxxxxxxxxx /d:sonar.verbose=true"
========================== Starting Command Output ===========================
##[command]"C:\windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "d:\a\_temp\8195ad81-4359-4594-ab9b-7d79d139b92a.cmd""
SonarScanner for MSBuild 4.6.1
Using the .NET Core version of the Scanner for MSBuild
Post-processing started.
10:15:02.897  Fetching code coverage report information from TFS...
10:15:02.899  Attempting to locate a test results (.trx) file...
10:15:02.909  Looking for TRX files in: d:\a\1\TestResults
10:15:02.91  No test results files found
10:15:02.914  Did not find any binary coverage files in the expected location.
10:15:02.915  Falling back on locating coverage files in the agent temp directory.
10:15:02.916  Searching for coverage files in d:\a\_temp
WARNING: Duplicate ProjectGuid: "387a19a1-842f-4048-91f6-09f37abba69f". The project will not be analyzed by SonarQube. Project file: "D:\a\1\s\src\Hocon.Json\Hocon.Json.csproj"
WARNING: Duplicate ProjectGuid: "387a19a1-842f-4048-91f6-09f37abba69f". The project will not be analyzed by SonarQube. Project file: "d:\a\1\s\src\Hocon.Json\Hocon.Json.csproj"
WARNING: Duplicate ProjectGuid: "642ca221-c030-4c2a-9f54-7c04c96ea95e". The project will not be analyzed by SonarQube. Project file: "D:\a\1\s\src\Hocon.Json.Tests\Hocon.Json.Tests.csproj"
WARNING: Duplicate ProjectGuid: "642ca221-c030-4c2a-9f54-7c04c96ea95e". The project will not be analyzed by SonarQube. Project file: "d:\a\1\s\src\Hocon.Json.Tests\Hocon.Json.Tests.csproj"
##[error]No analysable projects were found. SonarQube analysis will not be performed. Check the build summary report for details.
Generation of the sonar-properties file failed. Unable to complete SonarQube analysis.
##[error]10:15:03.056  Post-processing failed. Exit code: 1

##[error]Cmd.exe exited with code '1'.
##[section]Finishing: SonarScanner end
```

### Workaround
Pass the `--no-build` argument to `dotnet test`.
This will prevent projects from being built a second time and avoid the duplicate GUID issue.

### Root cause
When looking for duplicate projects, the scanner is doing a case-sensitive file path comparison, so `C:\myprop.csproj` and `c:\myproj.csproj` are treated as different projects.

Running `dotnet-sonarscanner begin`, `dotnet build ...`, `dotnet test ...`, `dotnet-sonarscanner end` from the command line works fine.
However, running exactly the same commands in an Azure Pipelines causes the above problem because the casing of the project file path in the `ProjectInfo.xml` file that is produced by `dotnet test` is not the same as that produced by `dotnet build` e.g.

ProjectInfo.xml (1)
```
  <ProjectGuid>387a19a1-842f-4048-91f6-09f37abba69f</ProjectGuid>
  <FullPath>D:\agents\test-ag\_work\6\s\src\Hocon.Json\Hocon.Json.csproj</FullPath>
```

ProjectInfo.xml (2)
```
  <ProjectGuid>387a19a1-842f-4048-91f6-09f37abba69f</ProjectGuid>
  <FullPath>d:\agents\test-ag\_work\6\s\src\Hocon.Json\Hocon.Json.csproj</FullPath>
```
For some reason, the casing of the drive letter is different. This is enough to make the scanner think there are two different projects with the same GUID.

### TODO
The simple fix would be to make the comparison case-insensitive. However, although Windows is a case-insensitive file system, Unix isn't. Decide whether it is necessary in practice to take the OS into account.

--- a/Tests/SonarScanner.MSBuild.Shim.Tests/PropertiesFileGeneratorTests.cs
+++ b/Tests/SonarScanner.MSBuild.Shim.Tests/PropertiesFileGeneratorTests.cs
@@ -99,8 +99,7 @@ namespace SonarScanner.MSBuild.Shim.Tests
             var testDir = TestUtils.CreateTestSpecificFolder(TestContext);
 
             var mockRuntimeInformation = new Mock<IRuntimeInformationWrapper>();
-            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.OSX)).Returns(true);
-            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.Linux)).Returns(false);
+            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.Windows)).Returns(false);
 
             var guid = Guid.NewGuid();
 
@@ -136,8 +135,7 @@ namespace SonarScanner.MSBuild.Shim.Tests
             var testDir = TestUtils.CreateTestSpecificFolder(TestContext);
 
             var mockRuntimeInformation = new Mock<IRuntimeInformationWrapper>();
-            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.OSX)).Returns(false);
-            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.Linux)).Returns(false);
+            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.Windows)).Returns(true);
 
             var guid = Guid.NewGuid();
 

--- a/Tests/SonarScanner.MSBuild.Shim.Tests/PropertiesFileGeneratorTests.cs
+++ b/Tests/SonarScanner.MSBuild.Shim.Tests/PropertiesFileGeneratorTests.cs
@@ -24,6 +24,7 @@ using System.IO;
 using System.Linq;
 using FluentAssertions;
 using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
 using SonarScanner.MSBuild.Common;
 using TestUtilities;
 
@@ -90,6 +91,72 @@ namespace SonarScanner.MSBuild.Shim.Tests
         }
 
         [TestMethod]
+        public void FileGen_Duplicate_SameGuid_DifferentCase_ShouldNotIgnoreCase()
+        {
+            // Casing can be ignored on windows OS
+
+            // Arrange
+            var testDir = TestUtils.CreateTestSpecificFolder(TestContext);
+
+            var mockRuntimeInformation = new Mock<IRuntimeInformationWrapper>();
+            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.OSX)).Returns(true);
+            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.Linux)).Returns(false);
+
+            var guid = Guid.NewGuid();
+
+            CreateProjectInfoInSubDir(testDir, "withoutFiles", guid, ProjectType.Product, false, "c:\\abc\\withoutfile.proj", "UTF-8");
+            CreateProjectInfoInSubDir(testDir, "withoutFiles1", guid, ProjectType.Product, false, "C:\\abc\\withoutFile.proj", "UTF-8"); // not excluded
+
+            var logger = new TestLogger();
+            var config = CreateValidConfig(testDir);
+
+            // Act
+            var result = new PropertiesFileGenerator(config, logger, new RoslynV1SarifFixer(logger), mockRuntimeInformation.Object).GenerateFile();
+
+            // Assert
+            AssertExpectedStatus("withoutFiles", ProjectInfoValidity.DuplicateGuid, result);
+            AssertExpectedProjectCount(1, result);
+
+            logger.Warnings.Should().HaveCount(2);
+
+            logger.Warnings.Should().BeEquivalentTo(
+               new[]
+               {
+                    $"Duplicate ProjectGuid: \"{guid}\". The project will not be analyzed by SonarQube. Project file: \"c:\\abc\\withoutfile.proj\"",
+                    $"Duplicate ProjectGuid: \"{guid}\". The project will not be analyzed by SonarQube. Project file: \"C:\\abc\\withoutFile.proj\"",
+               });
+        }
+
+        [TestMethod]
+        public void FileGen_Duplicate_SameGuid_DifferentCase_ShouldIgnoreCase()
+        {
+            // Casing can be ignored on windows OS
+
+            // Arrange
+            var testDir = TestUtils.CreateTestSpecificFolder(TestContext);
+
+            var mockRuntimeInformation = new Mock<IRuntimeInformationWrapper>();
+            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.OSX)).Returns(false);
+            mockRuntimeInformation.Setup(m => m.IsOS(System.Runtime.InteropServices.OSPlatform.Linux)).Returns(false);
+
+            var guid = Guid.NewGuid();
+
+            CreateProjectInfoInSubDir(testDir, "withoutFiles", guid, ProjectType.Product, false, "c:\\abc\\withoutfile.proj", "UTF-8");
+            CreateProjectInfoInSubDir(testDir, "withoutFiles1", guid, ProjectType.Product, false, "C:\\abc\\withoutFile.proj", "UTF-8"); // not excluded
+
+            var logger = new TestLogger();
+            var config = CreateValidConfig(testDir);
+
+            // Act
+            var result = new PropertiesFileGenerator(config, logger, new RoslynV1SarifFixer(logger), mockRuntimeInformation.Object).GenerateFile();
+
+            // Assert
+            AssertExpectedStatus("withoutFiles", ProjectInfoValidity.NoFilesToAnalyze, result);
+            AssertExpectedProjectCount(1, result);
+
+        }
+
+        [TestMethod]
         public void FileGen_ValidFiles_SourceEncoding_Provided()
         {
             // Arrange
@@ -132,7 +199,7 @@ namespace SonarScanner.MSBuild.Shim.Tests
             };
 
             // Act
-            var result = new PropertiesFileGenerator(config, logger).GenerateFile();
+            var result = new PropertiesFileGenerator(config, logger, new RoslynV1SarifFixer(logger), new RuntimeInformationWrapper()).GenerateFile();
 
             // Assert
             var settingsFileContent = File.ReadAllText(result.FullPropertiesFilePath);
@@ -166,10 +233,10 @@ namespace SonarScanner.MSBuild.Shim.Tests
             var mockReturnPath = mockSarifFixer.ReturnVal;
 
             // Act
-            var result = new PropertiesFileGenerator(config, logger, mockSarifFixer).GenerateFile();
+            var result = new PropertiesFileGenerator(config, logger, mockSarifFixer, new RuntimeInformationWrapper()).GenerateFile();
 
             // Assert
-             mockSarifFixer.CallCount.Should().Be(1);
+            mockSarifFixer.CallCount.Should().Be(1);
 
             // Already valid SARIF -> no change in file -> unchanged property
             var provider = new SQPropertiesFileReader(result.FullPropertiesFilePath);
@@ -204,11 +271,11 @@ namespace SonarScanner.MSBuild.Shim.Tests
             var escapedMockReturnPath = mockSarifFixer.ReturnVal.Replace(@"\", @"\\");
 
             // Act
-            var result = new PropertiesFileGenerator(config, logger, mockSarifFixer).GenerateFile();
+            var result = new PropertiesFileGenerator(config, logger, mockSarifFixer, new RuntimeInformationWrapper()).GenerateFile();
 
             // Assert
-             mockSarifFixer.CallCount.Should().Be(1);
-             mockSarifFixer.LastLanguage.Should().Be(RoslynV1SarifFixer.CSharpLanguage);
+            mockSarifFixer.CallCount.Should().Be(1);
+            mockSarifFixer.LastLanguage.Should().Be(RoslynV1SarifFixer.CSharpLanguage);
 
             // Fixable SARIF -> new file saved -> changed property
             var provider = new SQPropertiesFileReader(result.FullPropertiesFilePath);
@@ -243,11 +310,11 @@ namespace SonarScanner.MSBuild.Shim.Tests
             var escapedMockReturnPath = mockSarifFixer.ReturnVal.Replace(@"\", @"\\");
 
             // Act
-            var result = new PropertiesFileGenerator(config, logger, mockSarifFixer).GenerateFile();
+            var result = new PropertiesFileGenerator(config, logger, mockSarifFixer, new RuntimeInformationWrapper()).GenerateFile();
 
             // Assert
-             mockSarifFixer.CallCount.Should().Be(1);
-             mockSarifFixer.LastLanguage.Should().Be(RoslynV1SarifFixer.VBNetLanguage);
+            mockSarifFixer.CallCount.Should().Be(1);
+            mockSarifFixer.LastLanguage.Should().Be(RoslynV1SarifFixer.VBNetLanguage);
 
             // Fixable SARIF -> new file saved -> changed property
             var provider = new SQPropertiesFileReader(result.FullPropertiesFilePath);
@@ -278,10 +345,10 @@ namespace SonarScanner.MSBuild.Shim.Tests
             var mockSarifFixer = new MockRoslynV1SarifFixer(null);
 
             // Act
-            var result = new PropertiesFileGenerator(config, logger, mockSarifFixer).GenerateFile();
+            var result = new PropertiesFileGenerator(config, logger, mockSarifFixer, new RuntimeInformationWrapper()).GenerateFile();
 
             // Assert
-             mockSarifFixer.CallCount.Should().Be(1);
+            mockSarifFixer.CallCount.Should().Be(1);
 
             // One valid project info file -> file created
             AssertPropertiesFilesCreated(result, logger);
@@ -743,8 +810,10 @@ namespace SonarScanner.MSBuild.Shim.Tests
                 },
             };
 
+            var logger = new TestLogger();
+
             var analysisRootDir = TestUtils.CreateTestSpecificFolder(TestContext, "project");
-            var propertiesFileGenerator = new PropertiesFileGenerator(CreateValidConfig(analysisRootDir), new TestLogger());
+            var propertiesFileGenerator = new PropertiesFileGenerator(CreateValidConfig(analysisRootDir), logger, new RoslynV1SarifFixer(logger), new RuntimeInformationWrapper());
             var results = propertiesFileGenerator.ToProjectData(projectInfos.GroupBy(p => p.ProjectGuid).First()).AnalyzerOutPaths.ToList();
 
             results.Should().HaveCount(4);
@@ -753,6 +822,7 @@ namespace SonarScanner.MSBuild.Shim.Tests
             results[2].FullName.Should().Be(new FileInfo("4").FullName);
             results[3].FullName.Should().Be(new FileInfo("1").FullName);
         }
+
 
         [TestMethod]
         public void ToProjectData_ProjectsWithDuplicateGuid()

--- a/Tests/SonarScanner.MSBuild.Shim.Tests/SonarScanner.MSBuild.Shim.Tests.csproj
+++ b/Tests/SonarScanner.MSBuild.Shim.Tests/SonarScanner.MSBuild.Shim.Tests.csproj
@@ -47,6 +47,7 @@
     <Reference Include="Moq, Version=4.10.0.0, Culture=neutral, PublicKeyToken=69f491c39445e920, processorArchitecture=MSIL">
       <HintPath>..\..\packages\Moq.4.10.0\lib\net45\Moq.dll</HintPath>
     </Reference>
+    <Reference Include="netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL" />
     <!--
      See https://github.com/dotnet/sdk/issues/901
      This project is old-style but references a project with new-style packaging (i.e. that uses PackageReference).
@@ -58,6 +59,9 @@
     </Reference>
     <Reference Include="System" />
     <Reference Include="System.Configuration" />
+    <Reference Include="System.Runtime.InteropServices.RuntimeInformation, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\System.Runtime.InteropServices.RuntimeInformation.4.3.0\lib\net45\System.Runtime.InteropServices.RuntimeInformation.dll</HintPath>
+    </Reference>
     <Reference Include="System.Threading.Tasks.Extensions, Version=4.1.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL">
       <HintPath>..\..\packages\System.Threading.Tasks.Extensions.4.3.0\lib\portable-net45+win8+wp8+wpa81\System.Threading.Tasks.Extensions.dll</HintPath>
     </Reference>

--- a/Tests/SonarScanner.MSBuild.Shim.Tests/packages.config
+++ b/Tests/SonarScanner.MSBuild.Shim.Tests/packages.config
@@ -5,6 +5,7 @@
   <package id="FluentAssertions.Analyzers" version="0.11.4" targetFramework="net461" />
   <package id="Moq" version="4.10.0" targetFramework="net461" />
   <package id="Newtonsoft.Json" version="10.0.3" targetFramework="net461" />
+  <package id="System.Runtime.InteropServices.RuntimeInformation" version="4.3.0" targetFramework="net461" />
   <package id="System.Threading.Tasks.Extensions" version="4.3.0" targetFramework="net461" />
   <package id="System.ValueTuple" version="4.4.0" targetFramework="net461" />
 </packages>

--- a/src/Packaging/Packaging.csproj
+++ b/src/Packaging/Packaging.csproj
@@ -72,6 +72,7 @@
 
     <!-- Third-parties dlls -->
     <ScannerFiles Include="$(SourcesRoot)\SonarScanner.MSBuild.TFS.Classic\bin\$(Configuration)\$(ScannerNetFxVersion)\Microsoft.VisualStudio.Setup.Configuration.Interop.dll" />
+	  <ScannerFiles Include="$(SourcesRoot)\SonarScanner.MSBuild\bin\$(Configuration)\$(ScannerNetFxVersion)\System.Runtime.InteropServices.RuntimeInformation.dll" /> 
     <ScannerFiles Include="$(SourcesRoot)\SonarScanner.MSBuild\bin\$(Configuration)\$(ScannerNetFxVersion)\System.ValueTuple.dll" />
 
     <!-- Entry points -->

--- a/src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs
+++ b/src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs
@@ -276,8 +276,7 @@ namespace SonarScanner.MSBuild.Shim
             };
 
             // Find projects with different paths within the same group
-
-            var projectPathsInGroup = new List<string>();
+            List<string> projectPathsInGroup = null;
 
             if (runtimeInformationWrapper.IsOS(OSPlatform.Windows))
             {

--- a/src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs
+++ b/src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs
@@ -279,7 +279,7 @@ namespace SonarScanner.MSBuild.Shim
 
             var projectPathsInGroup = new List<string>();
 
-            if (!runtimeInformationWrapper.IsOS(OSPlatform.Linux) && !runtimeInformationWrapper.IsOS(OSPlatform.OSX))
+            if (runtimeInformationWrapper.IsOS(OSPlatform.Windows))
             {
                 projectPathsInGroup = projectsGroupedByGuid
                 .Select(x => x.FullPath?.ToLowerInvariant())

--- a/src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs
+++ b/src/SonarScanner.MSBuild.Shim/PropertiesFileGenerator.cs
@@ -22,6 +22,7 @@ using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Runtime.InteropServices;
 using System.Text;
 using SonarScanner.MSBuild.Common;
 using SonarScanner.MSBuild.Common.Interfaces;
@@ -42,17 +43,19 @@ namespace SonarScanner.MSBuild.Shim
         private readonly AnalysisConfig analysisConfig;
         private readonly ILogger logger;
         private readonly IRoslynV1SarifFixer fixer;
+        private readonly IRuntimeInformationWrapper runtimeInformationWrapper;
 
         public /*for testing*/ PropertiesFileGenerator(AnalysisConfig analysisConfig, ILogger logger,
-            IRoslynV1SarifFixer fixer)
+            IRoslynV1SarifFixer fixer, IRuntimeInformationWrapper runtimeInformationWrapper)
         {
             this.analysisConfig = analysisConfig ?? throw new ArgumentNullException(nameof(analysisConfig));
             this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
             this.fixer = fixer ?? throw new ArgumentNullException(nameof(fixer));
+            this.runtimeInformationWrapper = runtimeInformationWrapper ?? throw new ArgumentNullException(nameof(runtimeInformationWrapper));
         }
 
         public PropertiesFileGenerator(AnalysisConfig analysisConfig, ILogger logger)
-            : this(analysisConfig, logger, new RoslynV1SarifFixer(logger))
+            : this(analysisConfig, logger, new RoslynV1SarifFixer(logger), new RuntimeInformationWrapper())
         {
         }
 
@@ -273,10 +276,24 @@ namespace SonarScanner.MSBuild.Shim
             };
 
             // Find projects with different paths within the same group
-            var projectPathsInGroup = projectsGroupedByGuid
+
+            var projectPathsInGroup = new List<string>();
+
+            if (!runtimeInformationWrapper.IsOS(OSPlatform.Linux) && !runtimeInformationWrapper.IsOS(OSPlatform.OSX))
+            {
+                projectPathsInGroup = projectsGroupedByGuid
+                .Select(x => x.FullPath?.ToLowerInvariant())
+                .Distinct()
+                .ToList();
+            }
+            else
+            {
+                projectPathsInGroup = projectsGroupedByGuid
                 .Select(x => x.FullPath)
                 .Distinct()
                 .ToList();
+            }
+
             if (projectPathsInGroup.Count > 1)
             {
                 projectData.Status = ProjectInfoValidity.DuplicateGuid;

--- a/src/SonarScanner.MSBuild.Shim/RuntimeInformationWrapper.cs
+++ b/src/SonarScanner.MSBuild.Shim/RuntimeInformationWrapper.cs
@@ -1,4 +1,24 @@
-﻿#if NET46
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#if NET46
 using System.Runtime.InteropServices;
 #endif
 

--- a/src/SonarScanner.MSBuild.Shim/RuntimeInformationWrapper.cs
+++ b/src/SonarScanner.MSBuild.Shim/RuntimeInformationWrapper.cs
@@ -1,0 +1,20 @@
+﻿#if NET46
+using System.Runtime.InteropServices;
+#endif
+
+namespace SonarScanner.MSBuild.Shim
+{
+    /* Class for ease mocking in UT */
+    public class RuntimeInformationWrapper : IRuntimeInformationWrapper
+    {
+        public bool IsOS(System.Runtime.InteropServices.OSPlatform osPlatform)
+        {
+            return System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(osPlatform);
+        }
+    }
+
+    public interface IRuntimeInformationWrapper
+    {
+        bool IsOS(System.Runtime.InteropServices.OSPlatform osPlatform);
+    }
+}

--- a/src/SonarScanner.MSBuild.Shim/SonarScanner.MSBuild.Shim.csproj
+++ b/src/SonarScanner.MSBuild.Shim/SonarScanner.MSBuild.Shim.csproj
@@ -8,6 +8,7 @@
   </ItemGroup>
   <ItemGroup>
     <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
+    <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.3.0" />
     <PackageReference Include="System.ValueTuple" Version="4.4.0" />
   </ItemGroup>
   <ItemGroup>

--- a/src/SonarScanner.MSBuild.Tasks/SonarScanner.MSBuild.Tasks.csproj
+++ b/src/SonarScanner.MSBuild.Tasks/SonarScanner.MSBuild.Tasks.csproj
@@ -17,6 +17,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Build" Version="14.3.0" />
     <PackageReference Include="Microsoft.Build.Utilities.Core" Version="14.3.0" />
+    <PackageReference Include="System.Runtime.InteropServices.RuntimeInformation" Version="4.3.0" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\SonarScanner.MSBuild.Common\SonarScanner.MSBuild.Common.csproj" />

